src/main/java/ttt/WithinGridBoundaryValidator.java

```diff
@@ -0,0 +1,25 @@
+package ttt;
+
+public class WithinGridBoundaryValidator implements InputValidator {
+    private final Board board;
+
+    public WithinGridBoundaryValidator(Board board) {
+        this.board = board;
+    }
+
+    @Override
+    public ValidationResult isValid(String input) {
+        if (board.isWithinGridBoundary(zeroIndexed(input))) {
+            return new ValidationResult(input, true, "");
+        }
+        return new ValidationResult(input, false, invalidReason(input));
+    }
+
+    private int zeroIndexed(String input) {
```

Comment:
Translating to zero based index takes place in 2 validators and the prompt (when a valid input is made, and passed back to the Game). 

In order to display an error message here which echo's the incorrect input given, I can't see a way around this. As mentioned in the blog post, I can rip out the decrementing into a collaborator and inject it in to the two validators and prompt, but not sure if that is overkill.